Assets/Scripts/Framework/Train/TrainEngine.cs

```diff
@@ -218,5 +223,7 @@ private void UpdateBraking()
             _isBraking = false;
             _brakeTimer = 0f;
         }
+
+        private void OnSpeedStateChange(int previousValue, int newValue, bool asServer) => OnSpeedStateChanged?.Invoke(newValue);
```

Comment:
```suggestion
private void OnSpeedStateChange(int previousValue, int newValue, bool asServer) 
        => OnSpeedStateChanged?.Invoke(newValue);
```

Line exceeds 120 characters